includes/lcd.h

```diff
@@ -0,0 +1,69 @@
+#ifndef LCD_H
+#define LCD_H
+
+#include "gpio.h"
+#include "printf.h"
+
+#include <stdint.h>
+
+typedef struct  LCD_handler_s
+{
+    GPIO_pin    *rs_pin; // LOW: command.  HIGH: character.
+    GPIO_pin    *enable_pin; // activated by a HIGH pulse.
+    GPIO_pin    *data_pins[4];
+
+    uint8_t     displayfunction;
+    uint8_t     displaycontrol;
+    uint8_t     displaymode;
+
+    uint8_t     numlines;
+    uint8_t     row_offsets[4];
+}               LCD_handler;
+
+// Call this function before anything esle
```

Comment:
It is convoluted that the caller should fill in `rs_pin`, `enable_pin`, and `data_pins` but not other fields (and this is not documented).